packages/api/src/controllers/helpers.test.ts

```diff
@@ -17,53 +21,164 @@ afterEach(async () => {
+      );
+      expect(presignedUrl).not.toContain("secretpwd");
     });
-    return getS3PresignedUrl(os, file);
-  };
-
-  it("should support old object store URLs with region", async () => {
-    const presignedUrl = await makePresignedUrl(
-      "s3+http://localhost:8000/test-region/test-bucket"
-    );
-    expect(presignedUrl).toMatch(
-      /^http:\/\/localhost:8000\/test-bucket\/test\.txt.+test-region.+$/
-    );
-  });
 
-  it("should support new object store URLs without region", async () => {
-    const presignedUrl = await makePresignedUrl(
-      "s3+https://localhost:8000/test-bucket"
-    );
-    expect(presignedUrl).toMatch(
-      /^https:\/\/localhost:8000\/test-bucket\/test\.txt.+ignored.+$/
-    );
+    it("should NOT support invalid object store URLs", async () => {
+      await expect(
+        makePresignedUrl("not-s3://localhost:8000/test-bucket")
+      ).rejects.toThrow(/not-s3:/);
+      await expect(
+        makePresignedUrl("s3+https://localhost:8000/")
+      ).rejects.toThrow(/"\/"/);
+      await expect(
+        makePresignedUrl("s3+https://localhost:8000/region/bucket/path")
+      ).rejects.toThrow(/"\/region\/bucket\/path"/);
+    });
   });
 
-  it("should support access credentials", async () => {
-    const presignedUrl = await makePresignedUrl(
-      "s3+https://poweruser:secretpwd@localhost:8000/test-bucket"
-    );
-    expect(presignedUrl).toMatch(
-      /^https:\/\/localhost:8000\/test-bucket\/test\.txt.+poweruser.+$/
-    );
-    expect(presignedUrl).not.toContain("secretpwd");
+  describe("convert storage object to object store URL", () => {
+    it("should convert correct object", () => {
+      const storageObj = {
+        endpoint: "https://gateway.storjshare.io",
+        bucket: "testbucket",
+        credentials: {
+          accessKeyId: "AKIAIOSFODNN7EXAMPLE",
+          secretAccessKey: "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY",
+        },
+      };
+      expect(toObjectStoreUrl(storageObj)).toBe(
+        "s3+https://AKIAIOSFODNN7EXAMPLE:wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY@gateway.storjshare.io/testbucket"
+      );
+    });
+
+    it("should fail if endpoint is not defined", () => {
+      const storageObj = {
+        bucket: "testbucket",
+        credentials: {
+          accessKeyId: "AKIAIOSFODNN7EXAMPLE",
+          secretAccessKey: "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY",
+        },
+      };
+      expect(() => toObjectStoreUrl(storageObj)).toThrow(
+        "undefined property 'endpoint'"
+      );
+    });
+
+    it("should fail if endpoint is empty", () => {
+      const storageObj = {
+        endpoint: "",
+        bucket: "testbucket",
+        credentials: {
+          accessKeyId: "AKIAIOSFODNN7EXAMPLE",
+          secretAccessKey: "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY",
+        },
+      };
+      expect(() => toObjectStoreUrl(storageObj)).toThrow(
+        "undefined property 'endpoint'"
+      );
```

Comment:
Since out target now is Storj I'd make the endpoint mandatory. Technically, right, we could say that it works with every S3-compatible storage, but I think it'd require some minor additional work, like: (as you mentioned) adding region, testing with private AWS buckets, some doc, etc.

I think we're open to change it in future and use the default AWS endpoint if the `endpoint` param is not specified.